include/openmc/tallies/filter_match.h

```diff
@@ -22,6 +23,26 @@ class FilterMatch
   int bins_weights_length_ {0};
   int i_bin_;
   bool bins_present_ {false};
+
+  void push_back(int bin, double weight) {
+    if (bins_weights_length_ >= FILTERMATCH_BINS_WEIGHTS_SIZE ) {
```

Comment:
The check for the number of matches could probably be factored out here I think.

Reply:
One upside to leaving the check in is that if someone uses a model that exceeds the hit limit, there will be some notification that this is what broke. I think it's nice to give the feedback to the user so we can save time on debugging, but if there is a better way of checking for this (or if we are already checking it somewhere else) let me know!

Reply:
Oh, I think the check and message are great! I was just wondering if we might be able to put it into a separate method since the lines at the beginning of both `push_back` methods are the same.